apps/src/applab/Exporter.js

```diff
@@ -9,6 +9,136 @@ import elementLibrary from './designElements/library';
+    "useModalFunctionEditor": true,
+    "useContractEditor": true,
+    "contractHighlight": true,
+    "contractCollapse": true,
+    "examplesHighlight": true,
+    "examplesCollapse": true,
+    "definitionHighlight": true,
+    "definitionCollapse": true,
+    "freePlay": true,
+    "appWidth": true,
+    "appHeight": true,
+    "sliderSpeed": true,
+    "calloutJson": true,
+    "disableExamples": true,
+    "showTurtleBeforeRun": true,
+    "autocompletePaletteApisOnly": true,
+    "textModeAtStart": true,
+    "designModeAtStart": true,
+    "hideDesignMode": true,
+    "beginnerMode": true,
+    "levelId": true,
+    "puzzle_number": true,
+    "stage_total": true,
+    "iframeEmbed": true,
+    "lastAttempt": true,
+    "submittable": true
+  },
+  "showUnusedBlocks": true,
+  "fullWidth": true,
+  "noHeader": true,
+  "noFooter": true,
+  "smallFooter": true,
+  "codeStudioLogo": true,
+  "hasI18n": true,
+  "callouts": true,
+  "channel": true,
+  "readonlyWorkspace": true,
+  "isLegacyShare": true,
+  "postMilestone": true,
+  "postFinalMilestone": true,
+  "puzzleRatingsUrl": true,
+  "authoredHintViewRequestsUrl": true,
+  "serverLevelId": true,
+  "gameDisplayName": true,
+  "publicCaching": true,
+  "is13Plus": true,
+  "hasContainedLevels": true,
+  "hideSource": true,
+  "share": true,
+  "labUserId": true,
+  "firebaseName": true,
+  "firebaseAuthToken": true,
+  "firebaseChannelIdSuffix": true,
+  "isSignedIn": true,
+  "pinWorkspaceToBottom": true,
+  "hasVerticalScrollbars": true,
+  "showExampleTestButtons": true,
+  "rackEnv": true,
+  "report": {
+    "fallback_response": true,
+    "callback": true,
+    "sublevelCallback": true,
+  },
+  "sendToPhone": true,
+  "send_to_phone_url": true,
+  "copyrightStrings": {
+    "thank_you": true,
+    "help_from_html": true,
+    "art_from_html": true,
+    "code_from_html": true,
+    "powered_by_aws": true,
+    "trademark": true,
+  },
+  "teacherMarkdown": true,
+  "dialog": {
+    "skipSound": true,
+    "preTitle": true,
+    "fallbackResponse": true,
+    "callback": true,
+    "sublevelCallback": true,
+    "app": true,
+    "level": true,
+    "shouldShowDialog": true,
+  },
+  "locale": true,
+};
+
+export function getAppOptionsFile() {
+  function getAppOptionsAtPath(whitelist, sourceOptions) {
+    if (!whitelist) {
+      return null;
+    }
+    const options = {};
+    Object.keys(whitelist).forEach(key => {
+      if (sourceOptions[key] === undefined) {
+        return;
+      }
+      if (typeof whitelist[key] === 'object') {
+        const subOptions = getAppOptionsAtPath(whitelist[key], sourceOptions[key]);
+        if (subOptions !== null) {
+          options[key] = subOptions;
+        }
+      } else {
+        options[key] = sourceOptions[key];
+      }
+    });
+    return options;
+  }
+  const options = getAppOptionsAtPath(APP_OPTIONS_WHITELIST, getAppOptions());
+  return `window.APP_OPTIONS = ${JSON.stringify(options)};`;
+}
```

Comment:
This method could use a unit test, should document the expected format of the `whitelist` argument, and could probably be simplified - especially if we can assume the whitelist grammar is strict (if you'll forgive the pseudo-BNF):

```
<whitelist> ::= Object.<string, value>
<value> ::= true | <whitelist>
```

Couldn't the inner method reduce to something like this?

```javascript
  function getAppOptionsAtPath(whitelist, sourceOptions) {
    if (!whitelist || !sourceOptions) {
      return null;
    }

    return _.reduce(whitelist, (memo, value, key) => {
      if (value === true) {
        memo[key] = sourceOptions[key];
      } else if (typeof value === 'object' && typeof sourceOptions[key] === 'object') {
        memo[key] = getAppOptionsAtPath(value, sourceOptions[key]);
      }
      return memo;
    }, {});
  }
```